cedars/app/api.py

```diff
@@ -0,0 +1,94 @@
+import os
+from loguru import logger
+from tenacity import retry, wait_exponential
+import requests
+from . import db
+
+def load_pines_url(project_id, superbio_api_token = None):
+    '''
+    if PINES_URL is not available in the ENV then
+    - Start a PINES SERVER
+    - With retry logic - keep making get requests
+    - Get request gives a PINES URL
+    - Call this URL for PINES predictions
+    '''
+
+    env_url = os.getenv("PINES_API_URL")
+    api_url = os.getenv("SUPERBIO_API_URL")
+    if env_url is not None:
+        # Get PINES api from .env
+        pines_api_url = env_url
+        is_url_from_api = False
+        logger.info(f"Received url : {pines_api_url} for pines from ENV variables.")
+    elif api_url is not None:
+        # Get PINES api from API
+        # Send a POST request to start the SERVER
+        endpoint = f"cedars_projects/{project_id}/pines"
+
+        if superbio_api_token is not None:
+            headers = {"Authorization": f"Bearer {superbio_api_token}"}
+        else:
+            logger.error("No API token found, cannot authenticate with the server.")
+            return None, False
+
+        try:
+            logger.info("\n\nPinging", f'{api_url}/{endpoint}')
+            logger.info("With header : ", headers, flush=True)
+            response = requests.post(f'{api_url}/{endpoint}', headers=headers, data={})
+            logger.info("POST responce", response, flush=True)
+
+            if response.status_code != 200:
+                raise requests.exceptions.HTTPError
+        except Exception as e:
+            logger.error(f"Encountered error {e} when trying to start PINES server")
```

Comment:
you should just `raise` the exception and put the calling function in try catch?